samcli/commands/check/lib/command_context.py

```diff
@@ -0,0 +1,92 @@
+"""
+A center hub for checker logic
+"""
+import os
+import functools
+import logging
+
+import boto3
+from boto3.session import Session
+
+from samtranslator.translator.managed_policy_translator import ManagedPolicyLoader
+from samtranslator.translator.translator import Translator
+from samtranslator.public.exceptions import InvalidDocumentException
+from samtranslator.parser import parser
+from samcli.commands.local.cli_common.user_exceptions import SamTemplateNotFoundException
+from samcli.yamlhelper import yaml_parse
+
+from samcli.lib.replace_uri.replace_uri import _replace_local_codeuri as external_replace_local_codeuri
+from ..exceptions import InvalidSamDocumentException
+
+LOG = logging.getLogger(__name__)
+
+
+class CheckContext:
+    """
+    This class translates a template (SAM or CFN json) into a CFN yaml format. Evenchually
+    this class will also contain the major function calls for sam check, such as
+    "ask_bottle_neck_quesions", "calculate_bottle_necks", "calculate_pricing", and
+    "print_results"
+    """
+
+    def __init__(self, region, profile, template_path):
+        self.region = region
+        self.profile = profile
+        self.template_path = template_path
+
+    def run(self):
+        self.transform_template()
+
+        LOG.info("... analyzing application template")
+
+    def transform_template(self):
+        """
+        Takes a sam template or a CFN json template and converts it into a CFN yaml template
```

Comment:
We technically do transform a CFN json to a CFN yaml. If a user wants to use a CFN json template, they can do so and `sam check` will still be able to read it, because it is transformed. CFN yaml is the end goal, regardless of whether the translator starts with a SAM or CFN json template

Reply:
my understanding is mainly our target here is to transform the SAM resources to CFN resources, and then we can do our checks on the CFN resources.